contracts/upgrades/Upgrade002.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "../utils/EIP2535Initializable.sol";
+import "../utils/IrrigationAccessControl.sol";
+import "../utils/EIP2535Initializable.sol";
+import "../core/WaterTowerStorage.sol";
+
+/// @title Upgrade 0.2: Add startTime on water tower pool storage
+
+contract Upgrade002 is EIP2535Initializable, IrrigationAccessControl {
+    using WaterTowerStorage for WaterTowerStorage.Layout;
+    struct OldPoolInfo {
+        uint256 totalRewardRate;
+        uint256 monthlyRewards;
+        uint256 endTime;
+    }
+
+    function init001() external reinitializer(2) onlyAdminRole {
+        WaterTowerStorage.Layout storage l = WaterTowerStorage.layout();
+        for (uint256 i = l.curPoolIndex; i > 0; --i) {
```

Comment:
Does this miss pool index 0?

Reply:
pool index 0 is not used, and it's endTime and startTime is 0